tasks/utilities/data_handling/task_fetch_srr_accession.wdl

```diff
@@ -0,0 +1,65 @@
+version 1.0
+
+task fetch_srr_accession {
+  input {
+    String sample_accession 
+    String docker = "us-docker.pkg.dev/general-theiagen/biocontainers/fastq-dl:2.0.4--pyhdfd78af_0"
+    Int disk_size = 10
+    Int cpu = 2
+    Int memory = 8
+  }
+  meta {
+    volatile: true
+  }
+
+    command <<< 
+    set -euo pipefail
+
+    # Output the current date and fastq-dl version for debugging
+    date -u | tee DATE
+    fastq-dl --version | tee VERSION
+
+    # Fetch metadata for the sample accession
+    echo "Fetching metadata for valid biosample ID or SRA: ~{sample_accession}"
+    if fastq-dl --accession ~{sample_accession} --only-download-metadata --verbose; then
+        if [[ -f fastq-run-info.tsv ]]; then
+            echo "Metadata written for valid biosample ID or SRA: ~{sample_accession}"
+            cat fastq-run-info.tsv
+
+            # Extract SRR accessions from the TSV file
+            SRR_accessions=$(awk -F'\t' 'NR>1 {print $1}' fastq-run-info.tsv | paste -sd ',' -)
+
+            if [[ -z "${SRR_accessions}" ]]; then
+                echo "No SRR accession found for valid biosample ID or SRA: ~{sample_accession}" > srr_accession.txt
+            else
+                echo "Extracted SRR accessions: ${SRR_accessions}"
+                echo "${SRR_accessions}" > srr_accession.txt
+            fi
+        else
+            echo "No metadata file found for valid biosample ID or SRA: ~{sample_accession}"
+            echo "No SRR accession found" > srr_accession.txt
+        fi
+    else
+        # Handle cases where fastq-dl exits with an error
```

Comment:
⚠️ this doesn't work as desired. I gave the workflow an invalid SRR ID and the workflow succeeded and said No SRR accession found.

Here's the difference in the log:

For a VALID SRR accession:
```
ERROR 2024-11-22 14:11:52:root:ERROR - fastq_dl.py:519
There was an issue querying ENA and
SRA, exiting...
ERROR 2024-11-22 14:11:52:root:ERROR - fastq_dl.py:520
STATUS: 200
ERROR 2024-11-22 14:11:52:root:ERROR - fastq_dl.py:521
TEXT: Query was successful, but
received an empty response
No results found for SAMD00010204
```

For an INVALID SRR accession:
```
2024-11-22 15:57:03 ERROR 2024-11-22 15:57:03:root:ERROR - fastq_dl.py:598
22-00052324 is not a Study, Sample,
Experiment, or Run accession. See
https://ena-docs.readthedocs.io/en/
latest/submit/general-guide/accessi
ons.html for valid options
No SRR accession found for valid biosample ID or SRA: 22-00052324
```

This is not desired behavior. Please adjust the logic in lines 44-50 to make sure that the workflow fails for invalid accessions but succeeds for valid accessions